artifacts/definitions/Linux/Remediation/Quarantine.yaml

```diff
@@ -0,0 +1,179 @@
+          combine_results(Stdout=Stdout, Stderr=Stderr,
+          ReturnCode=ReturnCode,
+          Message=TableName + ' added.') AS Result
+       FROM execve(argv=add_table_cmd, length=10000)
+
+     // add inbound chain
+     LET add_inbound_chain = SELECT
+          timestamp(epoch=now()) as Time,
+          combine_results(Stdout=Stdout, Stderr=Stderr,
+          ReturnCode=ReturnCode,
+          Message='Added inbound_chain to ' + TableName + ' table.') AS Result
+       FROM execve(argv=add_inbound_chain_cmd, length=10000)
+
+     // add tcp rule to inbound_chain to allow connections from Velociraptor
+     LET add_tcp_rule_to_inbound_chain = SELECT
+          timestamp(epoch=now()) as Time,
+          combine_results(Stdout=Stdout, Stderr=Stderr,
+          ReturnCode=ReturnCode,
+          Message='Added tcp rule to inbound_chain in ' + TableName + ' table.') AS Result
+       FROM execve(argv=add_tcp_rule_to_inbound_chain_cmd, length=10000)
+
+     // add udp rule to inbound_chain to allow DNS lookups
+     LET add_udp_rule_to_inbound_chain = SELECT
+          timestamp(epoch=now()) as Time,
+          combine_results(Stdout=Stdout, Stderr=Stderr,
+          ReturnCode=ReturnCode,
+          Message='Added udp rule to inbound_chain in ' + TableName + ' table.') AS Result
+       FROM execve(argv=add_udp_rule_to_inbound_chain_cmd, length=10000)
+
+     // add inbound chain
+     LET add_outbound_chain = SELECT
+          timestamp(epoch=now()) as Time,
+          combine_results(Stdout=Stdout, Stderr=Stderr,
+          ReturnCode=ReturnCode,
+          Message='Added outbound_chain to ' + TableName + ' table.') AS Result
+       FROM execve(argv=add_outbound_chain_cmd, length=10000)
+
+     // add tcp rule to outbound_chain to allow connections to Velociraptor
+     LET add_tcp_rule_to_outbound_chain = SELECT
+          timestamp(epoch=now()) as Time,
+          combine_results(Stdout=Stdout, Stderr=Stderr,
+          ReturnCode=ReturnCode,
+          Message='Added tcp rule to outbound_chain in ' + TableName + ' table.') AS Result
+       FROM execve(argv=add_tcp_rule_to_outbound_chain_cmd, length=10000)
+
+     // add udp rule to outbound_chain to allow DNS lookups
+     LET add_udp_rule_to_outbound_chain = SELECT
+          timestamp(epoch=now()) as Time,
+          combine_results(Stdout=Stdout, Stderr=Stderr,
+          ReturnCode=ReturnCode,
+          Message='Added udp rule to outbound_chain in ' + TableName + ' table.') AS Result
+       FROM execve(argv=add_udp_rule_to_outbound_chain_cmd, length=10000)
+
+     // add forward chain
+     LET add_forward_chain = SELECT
+          timestamp(epoch=now()) as Time,
+          combine_results(Stdout=Stdout, Stderr=Stderr,
+          ReturnCode=ReturnCode,
+          Message='Added forward_chain to ' + TableName + ' table.') AS Result
+       FROM execve(argv=add_forward_chain_cmd, length=10000)
+
+     // Execute content
+     SELECT * FROM if(condition=RemovePolicy,
```

Comment:
What happens if the rule accidentally cuts off access to the server? The Windows.Remediation.Quarantine artifact tests the connection and removes the quarantine if it can't reach the server. This seems like a good model to follow.

Reply:
Good point! Yes I agree we need to safeguard this. I'll borrow the Windows quarantine code.